vj4/util/validator.py

```diff
@@ -12,6 +12,8 @@
 ID_RE = re.compile(r'[^\\/\s\u3000]([^\\/\n\r]*[^\\/\s\u3000])?')
 ROLE_RE = re.compile(r'[_0-9A-Za-z]{1,256}')
 DOMAIN_INVITATION_CODE_RE = re.compile(r'[0-9A-Za-z]{1,64}')
+PID_RE1 = re.compile(r'[a-zA-Z0-9]*[a-zA-Z]+[a-zA-Z0-9]*')
```

Comment:
Too strange.

Reply:
This is for making sure that at least a letter is in string PID.

Reply:
Consider using the plan we just discuss: an ID `x` is a numeric PID if and only if `str(int(x)) == x` and no exception occurs during the evaluation.